activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -617,6 +618,32 @@ def order!(*args) # :nodoc:
     #   #     WHEN "users"."id" = 3 THEN 3
     #   #   END ASC
     #
+    # +column+ can point to an enum column; the actual query generated may be different depending
+    # on the database adapter and the column definition.
+    #
+    #   class Conversation < ActiveRecord::Base
+    #     enum :status, [ :active, :archived ]
+    #   end
+    #
+    #   Conversation.in_order_of(:status, %w(archived active))
+    #   # SELECT "conversations".* FROM "conversations"
+    #   #   WHERE "conversations"."status" IN (1, 0)
+    #   #   ORDER BY CASE
+    #   #     WHEN "conversations"."status" = 1 THEN 1
+    #   #     WHEN "conversations"."status" = 0 THEN 2
+    #   #   END ASC
+    #
+    # +values+ can also include +nil+.
+    #
+    #   Conversation.in_order_of(:status, %w(nil archived active))
```

Comment:
Not sure `nil` is working here as expected (it returns a String):
```ruby
%w(nil archived active)
=> ["nil", "archived", "active"]
```

Reply:
nice catch!